File: CVE/practice/tcc_tests.m
#import <AVFoundation/AVFoundation.h>

int main(void) {
    @autoreleasepool {
        AVAuthorizationStatus status =
            [AVCaptureDevice authorizationStatusForMediaType:AVMediaTypeAudio];

        NSLog(@"TCC status = %ld", (long)status);

        [AVCaptureDevice requestAccessForMediaType:AVMediaTypeAudio
                                  completionHandler:^(BOOL granted) {
            NSLog(@"granted = %d", granted);
            exit(0);
        }];

        [[NSRunLoop currentRunLoop] run];
    }

    return 0;
}

File: CVE/practice/tcc_tests.swift
import AVFoundation
import Foundation

let status = AVCaptureDevice.authorizationStatus(for: .audio)

print("TCC status = \(status.rawValue)")

AVCaptureDevice.requestAccess(for: .audio) { granted in
    print("granted = \(granted ? 1 : 0)")
    exit(0)
}

RunLoop.current.run()


